### PR TITLE
:sparkles: implement progpilot SAST parser #10044

### PR DESCRIPTION
**Pull request by manuel-sommer**
see issue #10044

### CODE REVIEW

**Review by Maffooch — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

**Review by grendel513 — APPROVED** [review]

File: unittests/tools/test_progpilot_parser.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from ..dojo_test_case import DojoTestCase
+from dojo.models import Test
+from dojo.tools.progpilot.parser import ProgpilotParser
+
+
+class TestProgpilotParser(DojoTestCase):
+
+    def test_crunch42parser_single_has_many_findings(self):
```
Comment (cneill) [inline unittests/tools/test_progpilot_parser.py:8]:
```suggestion
    def test_progpilotparser_single_has_many_findings(self):
```

File: unittests/tools/test_progpilot_parser.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            finding = findings[0]
+            self.assertEqual("Medium", finding.severity)
+            self.assertIsNotNone(finding.description)
+            self.assertGreater(len(finding.description), 0)
+            self.assertEqual(89, finding.cwe)
+            self.assertEqual("sql_injection", finding.title)
+            self.assertEqual("/home/User/Modules/progpilot/Order.php", finding.file_path)
+            self.assertEqual(593, finding.line)
+
+    def test_crunch42parser_single_has_one_finding(self):
```
Comment (cneill) [inline unittests/tools/test_progpilot_parser.py:24]:
```suggestion
    def test_progpilotparser_single_has_one_finding(self):
```

File: unittests/tools/test_progpilot_parser.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        testfile = open("unittests/scans/progpilot/progpilot2.json")
+        parser = ProgpilotParser()
+        findings = parser.get_findings(testfile, Test())
+        testfile.close()
+        self.assertEqual(1, len(findings))
+
+    def test_crunch42parser_single_has_many_findings3(self):
```
Comment (cneill) [inline unittests/tools/test_progpilot_parser.py:31]:
```suggestion
    def test_progpilotparser_single_has_many_findings3(self):
```

File: unittests/tools/test_progpilot_parser.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        testfile = open("unittests/scans/progpilot/progpilot3.json")
+        parser = ProgpilotParser()
+        findings = parser.get_findings(testfile, Test())
+        testfile.close()
+        self.assertEqual(3, len(findings))
+
+    def test_crunch42parser_single_has_many_findings4(self):
```
Comment (cneill) [inline unittests/tools/test_progpilot_parser.py:38]:
```suggestion
    def test_progpilotparser_single_has_many_findings4(self):
```

**Review by cneill — CHANGES_REQUESTED** [review]
Just a small typo, otherwise looks good

**Review by cneill — APPROVED** [review]

**Pull request merged**